### PR TITLE
Fix docker.save having `-default` in tar name

### PR DESCRIPTION
**Pull request by ericvn**
**Please provide a description of this PR:**

Fixes https://github.com/istio/istio/issues/36059

### CODE REVIEW

File: tools/docker-builder/main.go [hunk 1]
```diff
@@ -246,13 +246,13 @@ func ConstructBakeFile(a Args) (map[string]string, error) {
 				t.Outputs = []string{"type=registry"}
 			} else if args.Save {
 				n := target
-				if variant != "" {
+				if variant != "" && variant != DefaultVariant { // For default variant, we do not add it.
```
Comment (ericvn) [inline tools/docker-builder/main.go:249]: This line removes the `-default`.

File: tools/docker-builder/main.go [hunk 2]
```diff
@@ -246,13 +246,13 @@ func ConstructBakeFile(a Args) (map[string]string, error) {
...
 					n += "-" + variant
 				}
 
 				tarFiles[n] = ""
 				if variant == PrimaryVariant && hasDoubleDefault {
-					tarFiles[n] = target
+					tarFiles[n] = target + "-" + variant
```
Comment (ericvn) [inline tools/docker-builder/main.go:255]: We append the tag if doubled up. Using the alias to create another tar file. This line might not be needed or conflict with the alias elsewhere.

Comment (ericvn) [reply tools/docker-builder/main.go:255]: Looks like this change can be removed.

**Comment by ericvn** [comment]
What should `./tools/docker --save --variants=debug,distroless` generate for tar files. Should it generated bug, distroless and empty variant files? default,debug,distroless does generate all 3. It seems debug.distroless only creates debug and distroless and no empty variants.

**Review by jacob-delgado — APPROVED** [review]

**Comment by howardjohn** [comment]
> What should ./tools/docker --save --variants=debug,distroless generate for tar files. Should it generate debug, distroless and empty variant files? default,debug,distroless does generate all 3. It seems debug.distroless only creates debug and distroless and no empty variants. Testing prior to this change, I don't get the empty variant so this change doesn't change that.

It should generate debug and distroless ONLY.

**Pull request merged**